flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -300,18 +309,82 @@ public static boolean allocateResource(int[] nodeManagers, int toAllocate) {
 		return false;
 	}
 
-	@Override
 	public void setDetachedMode(boolean detachedMode) {
 		this.detached = detachedMode;
 	}
 
-	@Override
-	public boolean isDetached() {
+	public boolean isDetachedMode() {
 		return detached;
 	}
 
+
+	/**
+	 * Gets a Hadoop Yarn client
+	 * @return Returns a YarnClient which has to be shutdown manually
+	 */
+	private static YarnClient getYarnClient(Configuration conf) {
+		YarnClient yarnClient = YarnClient.createYarnClient();
+		yarnClient.init(conf);
+		yarnClient.start();
+		return yarnClient;
+	}
+
+	/**
+	 * Retrieves the Yarn application and cluster from the config
+	 * @param config The config with entries to retrieve the cluster
+	 * @return YarnClusterClient
+	 * @deprecated This should be removed in the future
+	 */
+	public YarnClusterClient retrieveFromConfig(org.apache.flink.configuration.Configuration config) throws  Exception {
+		String jobManagerHost = config.getString(ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY, null);
+		int jobManagerPort = config.getInteger(ConfigConstants.JOB_MANAGER_IPC_PORT_KEY, -1);
+
+		if (jobManagerHost != null && jobManagerPort != -1) {
+
+			YarnClient yarnClient = getYarnClient(conf);
+			List<ApplicationReport> applicationReports = yarnClient.getApplications();
+			for (ApplicationReport report : applicationReports) {
+				if (report.getHost().equals(jobManagerHost) && report.getRpcPort() == jobManagerPort) {
+					LOG.info("Found application '{}' " +
+						"with JobManager host name '{}' and port '{}' from Yarn properties file.",
+						report.getApplicationId(), jobManagerHost, jobManagerPort);
+					return retrieve(report.getApplicationId().toString());
+				}
+			}
+
```

Comment:
I wonder if it makes sense to log here that we were unable to find the application id?